trillian/util/log_leaf.go

```diff
@@ -71,3 +50,57 @@ func ExtraDataForChain(cert ct.ASN1Cert, chain []ct.ASN1Cert, isPrecert bool) ([
 	}
 	return tls.Marshal(extra)
 }
+
+func BuildLogLeafWithHash(logPrefix string, merkleLeaf ct.MerkleTreeLeaf, leafIndex int64, cert ct.ASN1Cert, chain []ct.ASN1Cert, chainHash []byte, isPrecert bool) (trillian.LogLeaf, error) {
+	return buildLogLeaf(logPrefix, merkleLeaf, leafIndex, cert, chain, chainHash, isPrecert)
+}
+
+// ExtraDataForChainWithHash creates the extra data associated with a log entry as
+// described in RFC6962 section 4.6.
+func ExtraDataForChainWithHash(cert ct.ASN1Cert, chain []ct.ASN1Cert, chainHash []byte, isPrecert bool) ([]byte, error) {
```

Comment:
Is `chain` used?

Reply:
Good catch. Removed the unused `chain` argument.